packages/app/src/background/services/zkIdentity/index.ts

```diff
@@ -407,22 +407,30 @@ export default class ZkIdentityService implements IBackupable {
   };
 
   downloadEncryptedStorage = async (backupPassword: string): Promise<string | null> => {
-    const backupEncryptedData = await this.identitiesStore.get<string>();
+    const data = await this.identitiesStore.get<string>();
 
-    if (!backupEncryptedData) {
+    if (!data) {
       return null;
     }
 
-    return this.cryptoService.generateEncryptedHmac(backupEncryptedData, backupPassword);
+    const backup = this.cryptoService.decrypt(data, { mode: ECryptMode.MNEMONIC });
+    const encryptedBackup = this.cryptoService.encrypt(backup, { secret: backupPassword });
+
+    return this.cryptoService.generateEncryptedHmac(encryptedBackup, backupPassword);
   };
 
   uploadEncryptedStorage = async (backupEncryptedData: string, backupPassword: string): Promise<void> => {
     if (!backupEncryptedData) {
       return;
     }
 
-    await this.identitiesStore.set<string>(
-      this.cryptoService.getAuthenticCiphertext(backupEncryptedData, backupPassword),
-    );
+    const encryptedBackup = this.cryptoService.getAuthenticCiphertext(backupEncryptedData, backupPassword);
+    const backup = this.cryptoService.decrypt(encryptedBackup, { secret: backupPassword });
+
+    const backupIdentities = new Map(JSON.parse(backup) as Iterable<readonly [string, string]>);
+    const identities = await this.getIdentitiesFromStore();
+    const mergedIdentities = new Map([...identities, ...backupIdentities]);
```

Comment:
**Future related thing**
Not sure if you considered that already or not, but do you think we should warn user with the common data between both the current  `identities` and the backup identities `backupIdentities` before doing overriding to the data?

Reply:
Just to note: the only one case with overriding is possible.

Steps to reproduce:
1. Install the extension
2. Save mnemonic
3. Create twitter identity with nonce 0
4. Rename identity to "Twitter"
5. Backup data
6. Reinstall extension using saved mnemonic from step 2
7. Create twitter identity with nonce 0 and keep the default name
8. Upload backup file

After this steps you'll see only one identity with name "Twitter" which saved from backup file. In this case all the metadata for current identity will be replaced with backup version.